database/rpcdb/db_test.go

```diff
@@ -112,15 +108,17 @@ func TestHealthCheck(t *testing.T) {
 	}
 	for _, scenario := range scenarios {
 		t.Run(scenario.name, func(t *testing.T) {
+			require := require.New(t)
+
 			baseDB := setupDB(t)
 			db := corruptabledb.New(baseDB.server)
 			defer db.Close()
 			require.NoError(scenario.testFn(db))
 
 			// check db HealthCheck
 			_, err := db.HealthCheck(context.Background())
-			if err == nil && scenario.wantErr {
-				t.Fatalf("wanted error got nil")
+			if err == nil {
+				require.False(scenario.wantErr)
```

Comment:
isn't the below condition already checking that? if we're not sure about `Containsf` checks this, we can add a `require.Error` just above that.

Reply:
Yeah, you're right. I thought that this was the best way to do it considering:
- we need to break on `err == nil` otherwise `err.Error()` will panic
- `require.Error` is banned in favor of `require.ErrorIs` in the codebase

Reply:
ah I see, maybe a check with `require.NotNil(err)` then?